iree/compiler/Conversion/Common/LinalgBufferizePass.cpp

```diff
@@ -1030,6 +1039,48 @@ static void copyFromAliasingBufferToResultBuffer(OpBuilder &b, Location loc,
   }
 }
 
+static LogicalResult convertPadTensorOp(
+    OpBuilder &b, linalg::PadTensorOp padTensorOp, BlockAndValueMapping &bvm,
+    BufferizationPlan &plan, WorkgroupMemoryAllocationFn allocationFn) {
+  auto inputTensor = padTensorOp.source();
+  auto inputMemref = bvm.lookup(inputTensor);
+
+  auto loc = padTensorOp.getLoc();
+
+  if (failed(getOrAllocateResultBuffers(b, padTensorOp, padTensorOp.result(),
+                                        bvm, plan, allocationFn))) {
+    return failure();
+  }
+
+  auto resultPaddedBuffer = bvm.lookup(padTensorOp.result());
+
+  // Get padding value and fill the result buffer.
+  linalg::YieldOp yeildOp =
+      *padTensorOp.region().getOps<linalg::YieldOp>().begin();
+  Value padValue = yeildOp.values()[0];
+  b.create<linalg::FillOp>(loc, resultPaddedBuffer, padValue);
+
+  // Get the interior region.
+  auto inputShape =
```

Comment:
Nit: Move the `sizeMixedValues` computation into a utility function. It seems like we might need this often.